### PR TITLE
Location Engine Singleton Bug

### PR DESCRIPTION
**Pull request by electrostat**
`obtainBestLocationEngine` returns the same instance of a location engine when called. This is causing issues when multiple location engines are used within an app (and created with this method) and getting out of sync with their desired behavior.

cc: @lilykaiser @zugaldia @Guardiola31337

### CODE REVIEW

File: liblocation/src/main/java/com/mapbox/android/core/location/LocationEngineProvider.java [hunk 1]
```diff
@@ -44,6 +45,7 @@ private void initAvailableLocationEngines(Context context) {
         locationEngineDictionary.put(entry.getKey(), available);
       }
     }
+    Log.e("dropped-location", "locationEngineDictionary: " + locationEngineDictionary);
```
Comment (zugaldia) [inline liblocation/src/main/java/com/mapbox/android/core/location/LocationEngineProvider.java]: We should probably remove these logs before merging.

File: app/src/main/java/com/mapbox/android/events/testapp/MainActivity.java [hunk 2]
```diff
@@ -20,7 +20,7 @@ protected void onCreate(Bundle savedInstanceState) {
     setContentView(R.layout.activity_main);
 
     String accessTokenTelemetry = obtainAccessToken();
-    String userAgentTelemetry = "MapboxEventsAndroid/3.1.0";
+    String userAgentTelemetry = "MapboxEventsAndroid/";
```
Comment (zugaldia) [inline app/src/main/java/com/mapbox/android/events/testapp/MainActivity.java]: This change seems unrelated to this issue?

File: liblocation/src/main/java/com/mapbox/android/core/location/LocationEngineProvider.java [hunk 3]
```diff
@@ -57,6 +57,7 @@ private void initAvailableLocationEngines(Context context) {
 
   private LocationEngine obtainBestLocationEngine() {
     LocationEngine androidLocationEngine = locationEngineDictionary.get(LocationEngine.Type.ANDROID);
+
```
Comment (Guardiola31337) [inline liblocation/src/main/java/com/mapbox/android/core/location/LocationEngineProvider.java]: NIT empty line
